### PR TITLE
Make header file inclusion platform independent

### PR DESCRIPTION
**Pull request by SudeshnaMoulik**
#### Describe Bug or Feature
Make header file inclusion platform independent.

#### Describe Your Change
 - Made the header file inclusion platform independent.
 - Remove dead code file: get_localaddr.c
- Made qmgr code platform independent for QMGR_HAVE_HIST macro
- Fix logging error in: resmom/requests.c

#### Attach Test and Valgrind Logs/Output

image

### CODE REVIEW

File: src/cmds/Makefile.am [hunk 1]
```diff
@@ -194,7 +194,7 @@ qmgr_CPPFLAGS = \
 qmgr_LDADD = \
 	${common_libs} \
 	@editline_lib@
-qmgr_SOURCES = qmgr.c ${common_sources}
+qmgr_SOURCES = qmgr.c qmgr_sup.c ${common_sources}
```
Comment (hirenvadalia) [inline src/cmds/Makefile.am:197]: is this change intentional?

Comment (SudeshnaMoulik) [reply src/cmds/Makefile.am:197]: Hi @hirenvadalia yes. We are bifurcating some part of code in qmgr, to make it platform independent.

File: src/include/server_limits.h [hunk 2]
```diff
@@ -42,9 +42,7 @@
 #ifdef	__cplusplus
 extern "C" {
 #endif
-#ifdef WIN32
-#include "win.h"
-#endif
+#include <pbs_config.h>
```
Comment (bayucan) [inline src/include/server_limits.h:45]: I only see a few places where removal of "win.h" (and sometimes winsock.h") is replaced with #include <pbs_config.h>. What determines this? Saw this also in src/lib/Libutil/entlim.c and src/tools/pbs_sleep.c, but not in:
src/resmom/job_recov_fs.c,
src/resmom/requests.c,
src/resmom/stage_func.c,
src/server/job_func.c
src/server/req_quejob.c

File: src/lib/Libifl/pbs_ifl_wrap.c [hunk 3]
```diff
@@ -139,12 +139,6 @@
 # pragma warning disable 592
 #endif
 
-
-#if defined(__GNUC__) && defined(_WIN32) && !defined(SWIG_PYTHON_NO_HYPOT_WORKAROUND)
-/* Workaround for '::hypot' has not been declared', see https://bugs.python.org/issue11566 */
-# include <math.h>
-#endif
-
```
Comment (bayucan) [inline src/lib/Libifl/pbs_ifl_wrap.c:147]: What's the motivation for removing this? I'm worried that  it might break something in our hooks feature. If it's because it matches WIN32, but windows mom and pbs_python still use Python...

File: src/include/server_limits.h (hunk 2 above)
Comment (SudeshnaMoulik) [reply src/include/server_limits.h:45]: Hi @bayucan Thanks for taking a look in this PR.
Scope of this PR is to move away from any specific header file inclusion depnending on WIN32.
For Windows both "win.h" and "winsock.h" is part of pbs_config.h now.
W didn't add that on those file, because its already there.

File: src/lib/Libifl/pbs_ifl_wrap.c (hunk 3 above)
Comment (SudeshnaMoulik) [reply src/lib/Libifl/pbs_ifl_wrap.c:147]: If I go through the issue https://bugs.python.org/issue11566 :
- Windows affect version is not mentioned as 3.6
- without adding this didn't face any compiler error
- ran execjob_epilogoe and execjob_prologue hook without any error
- apparently looks like issue was for "the Microsoft Visual C++ Compiler for Python 2.7."

Please let me know whether I should test few more things after removing that header ?

Comment (bayucan) [reply src/lib/Libifl/pbs_ifl_wrap.c:147]: So I looked further into this file, src/lib/Libifl/pbs_ifl_wrap.c, and the heading says:
/* ----------------------------------------------------------------------------
 * This file was automatically generated by SWIG (http://www.swig.org).
 * Version 4.0.1
 *
 * This file is not intended to be easily readable and contains a number of
 * coding conventions designed to improve portability and efficiency. Do not make
 * changes to this file unless you know what you are doing--modify the SWIG
 * interface file instead.
 * ----------------------------------------------------------------------------- */

This file is only used on Windows, when 'swig' is not available to create pbs_ifl_wrap.c on the fly. Otherwise, it is auto-created via make (and WIndows gen_ifl_wrap.bat script) as follows:
bayucan@pbi-18-s12:~/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/src/lib/Libpython> rm -i pbs_ifl_wrap.c
rm: remove regular file ‘pbs_ifl_wrap.c’? y
bayucan@pbi-18-s12:~/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/src/lib/Libpython> make
echo '%module pbs_ifl' > pbs_ifl.i ; \
echo '%{' >> pbs_ifl.i ; \
echo '#include "pbs_ifl.h"' >> pbs_ifl.i ; \
echo '%}' >> pbs_ifl.i ; \
echo '%include "pbs_ifl.h"' >> pbs_ifl.i ; \
/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/swig/bin/swig -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/swig/share/swig/3.0.10 -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/swig/share/swig/3.0.10/python \
        -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/include \
        -python pbs_ifl.i
gcc -DHAVE_CONFIG_H -I. -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/lib/Libpython -I../../../src/include  -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/include -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/python/include/python3.6m -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/python/include/python3.6m  -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/tcltk/include -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/alsdk/include  -g -Wall -Werror -Wno-unused-result -fPIC -D_LARGEFILE_SOURCE -O2     -MT libpbspython_a-pbs_ifl_wrap.o -MD -MP -MF .deps/libpbspython_a-pbs_ifl_wrap.Tpo -c -o libpbspython_a-pbs_ifl_wrap.o `test -f 'pbs_ifl_wrap.c' || echo '/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/lib/Libpython/'`pbs_ifl_wrap.c
mv -f .deps/libpbspython_a-pbs_ifl_wrap.Tpo .deps/libpbspython_a-pbs_ifl_wrap.Po
rm -f libpbspython.a
ar cru libpbspython.a libpbspython_a-shared_python_utils.o libpbspython_a-common_python_utils.o libpbspython_a-pbs_ifl_wrap.o libpbspython_a-pbs_python_external.o libpbspython_a-pbs_python_svr_external.o libpbspython_a-module_pbs_v1.o libpbspython_a-pbs_python_svr_internal.o libpbspython_a-pbs_python_svr_size_type.o libpbspython_a-pbs_python_import_types.o 
ranlib libpbspython.a
gcc -DHAVE_CONFIG_H -I. -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/lib/Libpython -I../../../src/include  -DLIBPYTHONSVR -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/include -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/python/include/python3.6m -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/python/include/python3.6m  -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/tcltk/include -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/alsdk/include  -g -Wall -Werror -Wno-unused-result -fPIC -D_LARGEFILE_SOURCE -O2     -MT libpbspython_svr_a-pbs_ifl_wrap.o -MD -MP -MF .deps/libpbspython_svr_a-pbs_ifl_wrap.Tpo -c -o libpbspython_svr_a-pbs_ifl_wrap.o `test -f 'pbs_ifl_wrap.c' || echo '/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/lib/Libpython/'`pbs_ifl_wrap.c
mv -f .deps/libpbspython_svr_a-pbs_ifl_wrap.Tpo .deps/libpbspython_svr_a-pbs_ifl_wrap.Po
rm -f libpbspython_svr.a
ar cru libpbspython_svr.a libpbspython_svr_a-shared_python_utils.o libpbspython_svr_a-common_python_utils.o libpbspython_svr_a-pbs_ifl_wrap.o libpbspython_svr_a-pbs_python_external.o libpbspython_svr_a-pbs_python_svr_external.o libpbspython_svr_a-module_pbs_v1.o libpbspython_svr_a-pbs_python_svr_internal.o libpbspython_svr_a-pbs_python_svr_size_type.o libpbspython_svr_a-pbs_python_import_types.o 
ranlib libpbspython_svr.a
bayucan@pbi-18-s12:~/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/src/lib/Libpython> ls -l pbs_ifl_wrap.c
-rw-r--r-- 1 bayucan agt 327924 Jul 29 10:22 pbs_ifl_wrap.c

---
So I believe you don't need to update this file.

File: src/include/server_limits.h (hunk 2 above)
Comment (bayucan) [reply src/include/server_limits.h:45]: Ok.

**Comment by SudeshnaMoulik** [comment]
Thanks @bayucan for reviewing this PR

File: src/lib/Libtpp/tpp_util.c [hunk 4]
```diff
@@ -63,10 +63,9 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
-#ifndef WIN32
 #include <netinet/tcp.h>
 #include <arpa/inet.h>
-#endif
+#include "avltree.h"
```
Comment (hirenvadalia) [inline src/lib/Libtpp/tpp_util.c:68]: This should be pbs_idx.h.

File: src/resmom/mom_main.c [hunk 5]
```diff
@@ -110,7 +106,8 @@
 #include	"libsec.h"
 #include	"pbs_ecl.h"
 #include	"pbs_internal.h"
-#ifndef	WIN32
+#include	"avltree.h"
```
Comment (hirenvadalia) [inline src/resmom/mom_main.c:109]: same as above, now we don't use avltree.h in application logic directly, it should be pbs_idx.h

**Comment by SudeshnaMoulik** [comment]
Latest logs:
image
